src/tovnode.ts

```diff
@@ -15,6 +24,21 @@ export function toVNode(node: Node, domApi?: DOMAPI): VNode {
     let i: number, n: number;
     const elmAttrs = node.attributes;
     const elmChildren = node.childNodes;
+
+    // get element's datasets
+    const datasets: Record<string, string> = {};
+    getAttrs(elmAttrs).forEach((attr) => {
+      if (attr.name.startsWith("data-")) {
+        datasets[attr.name.replace("data-", "")] = attr.value;
+      }
+    });
+
+    // remove attributes from datasets
+    getAttrs(elmAttrs).forEach((attr) => {
+      Object.keys(datasets).includes(attr.name.replace("data-", "")) &&
+        elmAttrs.removeNamedItem(attr.name);
+    });
```

Comment:
Creating a new array twice, just to use the `forEach` method is not really necessary. Please refactor this to use a single `for` loop. Also, instead of `replace` please use `slice`, we are only interested in removing stuff from the beginning of the string anyways